modal/_container_entrypoint.py

```diff
@@ -246,7 +246,7 @@ def worker_thread():
                     logger.exception(f"Exception raised by {_func} in DaemonizedThreadPool worker!")
                 self.inputs.task_done()
 
-        if self.spawned_workers < self.max_threads:
+        if self.spawned_workers < self.container_io_manager.get_input_concurrency():
```

Comment:
Since this function is dynamic there are races where you fetch an input but don't have enough threads. Let's simplify this to just use max threads that is the max concurrent inputs?

Reply:
Ideally this doesn't depend on io manager